packages/fast-components/src/design-system-provider/design-system-provider.ts

```diff
@@ -1,98 +1,271 @@
+            this.consumers.forEach(this.writeConsumerRecipeData);
         }
     }
 
-    @designSystemProperty
-    public backgroundColor: string = "#FFFFFF";
+    public suscribe(consumer: DesignSystemConsumer): void {
+        if (!this.consumers.has(consumer)) {
+            this.consumers.add(consumer);
+            this.writeConsumerRecipeData(consumer);
+        }
+    }
 
-    @designSystemProperty
-    public baseHeightMultiplier: number = 8;
+    public unsubscribe(consumer: DesignSystemConsumer): void {
+        this.consumers.delete(consumer);
+    }
 
-    @designSystemProperty
-    public bodyFont: string = "Segoe UI, sans-serif";
+    private writeConsumerRecipeData = (consumer: DesignSystemConsumer) => {
+        consumer.recipes.forEach(recipe => {
+            this.setCustomProperty({
+                name: recipe.name,
+                // use spread on the designSystem object to circumvent memoization
+                // done in the color recipes - we use the same *reference* in WC
+                // for performance improvements but that throws off the recipes
+                // We should look at making the recipes use simple args that
+                // we can individuall memoize.
+                value: recipe.resolver.bind(this, { ...this.designSystem }),
+            });
+        });
+    };
 
-    @designSystemProperty
-    public cornerRadius: number = 2;
+    /**
+     * Syncronizes the provider's design system with the local
+     * overrides. Any value defined on the instance will take priority
+     * over the value defined by the provider
+     */
+    private syncDesignSystemWithProvider(): void {
+        if (this.provider) {
+            Object.keys(this.provider.designSystem).forEach(key => {
+                if (this[key] === void 0) {
+                    this.designSystem[key] = this.provider!.designSystem[key];
+                }
+            });
+        }
+    }
 
-    @designSystemProperty
-    public disabledOpacity = 0.3;
+    /**
+     * Invoked when the provider observable property defined by the consumer is changed
+     * @param prev the previous value
+     * @param next the next value
+     */
+    private providerChanged(
+        prev: DesignSystemProvider | null,
+        next: DesignSystemProvider | null
+    ): void {
+        if (prev instanceof HTMLElement) {
+            Observable.getNotifier(prev).unsubscribe(this, designSystemKey);
+        }
 
-    @designSystemProperty
-    public density = 0;
+        if (next instanceof HTMLElement && isDesignSystemProvider(next)) {
+            Observable.getNotifier(next).subscribe(this, designSystemKey);
+            this.syncDesignSystemWithProvider();
+        }
+    }
+}
 
-    @designSystemProperty
-    public designUnit = 4;
+/**
+ * Simple proxy object to notifiy observers on
+ * object mutation.
+ * @param source The object to fire notifications
+ * @param key The observable property name to fire a notification on

```

Comment:
@EisenbergEffect this function is primarily a work-around for `@observable` because observable won't notify on property mutation. You mentioned something earlier on making observable support mutation I think, perhaps you have thoughts here.

Reply:
Each property of the object itself needs and observable decorator. Is it possible to change the source so that each property is an observable or an attr?

Reply:
I think it adds more overhead than it's worth. At this point we need to react when *any* change is made, it doesn't matter what it is. 

In the future I'd like to get us to a place where recipes depend on individual arguments and not the whole DS object so when we get there it might make sense to refactor this, but for now I'm going to keep this implementation to avoid complicating this component further.

Reply:
I think we can keep it for now. I bet we can come up for something in the future. I've been trying to avoid proxies so that we can at least say this can be polyfilled for IE11. In any case, let's go with this, since it does work, and then we'll circle back around later.

Reply:
Ah good point - okay I'll take another quick stab at this

Reply:
Got it to work